subprojects/parseq-guava-interop/src/main/java/com/linkedin/parseq/guava/ListenableFutureUtil.java

```diff
@@ -0,0 +1,133 @@
+package com.linkedin.parseq.guava;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AbstractFuture;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.linkedin.parseq.BaseTask;
+import com.linkedin.parseq.Context;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.promise.Promise;
+import com.linkedin.parseq.promise.SettablePromise;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+
+/**
+ * Utility methods to convert between Parseq {@link Task} and Guava's {@link ListenableFuture}.
+ */
+public class ListenableFutureUtil {
+
+  private ListenableFutureUtil() {
+    // Prevent instantiation.
+  }
+
+  public static <T> Task<T> fromListenableFuture(ListenableFuture<T> future) {
+    // Setup cancellation propagation from Task -> ListenableFuture.
+    final SettableTask<T> task =
+        new SettableTask<T>("fromListenableFuture: " + Task._taskDescriptor.getDescription(future.getClass().getName())) {
+          @Override
+          public boolean cancel(Exception rootReason) {
+            if (future.isCancelled()) {
+              return super.cancel(rootReason);
+            }
+
+            return super.cancel(rootReason) && future.cancel(true);
+          }
+        };
+
+    final SettablePromise<T> promise = task.getSettableDelegate();
+
+    // Setup forward event propagation ListenableFuture -> Task.
+    Runnable callbackRunnable = () -> {
+      try {
+        final T value = future.get();
+        promise.done(value);
+      } catch (CancellationException ex) {
+        task.cancel(ex);
+      } catch (ExecutionException ex) {
+        promise.fail(ex.getCause());
+      } catch (Exception | Error ex) {
+        promise.fail(ex);
+      }
+    };
+    future.addListener(callbackRunnable, MoreExecutors.directExecutor());
+
+    return task;
+  }
+
+  public static <T> ListenableFuture<T> toListenableFuture(Task<T> task) {
+    // Setup cancellation propagation from ListenableFuture -> Task.
+    SettableFuture<T> listenableFuture = new SettableFuture<T>() {
+      @Override
+      public boolean cancel(boolean mayInterruptIfRunning) {
+        return super.cancel(mayInterruptIfRunning) && task.cancel(new CancellationException());
+      }
+
+      @Override
+      public boolean setException(Throwable ex) {
+        if (!task.isDone() && ex instanceof CancellationException) {
+          task.cancel((CancellationException) ex);
+        }
+        return super.setException(ex);
+      }
+    };
+
+    // Setup forward event propagation Task -> ListenableFuture.
```

Comment:
nit: is the wording correct? or should it be backward (corresponding to the "forward" in "fromListenableFuture"

Reply:
I just find this clearer to read... and it is forward and backward right?